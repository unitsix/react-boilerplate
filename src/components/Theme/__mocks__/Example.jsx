
import React, { Component } from 'react'
import Theme from '../Theme'
import PropTypes from 'prop-types'

export default class Example extends Component {
  render = () => {
    const {children, dark = false} = this.props
    return (
      <Theme dark={dark}>
          {children}
      </Theme>
    )
  }
}

Example.propTypes = {
  children: PropTypes.any,
  dark: PropTypes.bool
}