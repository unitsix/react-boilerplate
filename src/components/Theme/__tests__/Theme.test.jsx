/* eslint-env jest */
import Example from '../__mocks__/Example'

describeComponent(Example, (component) => {
  it('renders in default mode', () => {
    component.defaultProps = { children: 'hello world' }
    expect(component.snapshot()).toMatchSnapshot()
  })
  it('renders in dark mode', () => {
    component.defaultProps = { children: 'hello world', dark: true }
    expect(component.snapshot()).toMatchSnapshot()
  })
})
