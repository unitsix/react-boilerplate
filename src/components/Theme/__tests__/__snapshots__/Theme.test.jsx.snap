// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Example /> renders in dark mode 1`] = `
<div
  className="sc-AxjAm kJwmHj"
>
  hello world
</div>
`;

exports[`Component: <Example /> renders in default mode 1`] = `
<div
  className="sc-AxjAm kJwmHj"
>
  hello world
</div>
`;
